src/graph/components/app.tsx

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import { BaseApp } from "../../shared/components/base-app";
+import { IAuthoredState } from "./types";
+import { Runtime } from "./runtime";
+import { JSONSchema6 } from "json-schema";
+
+const baseAuthoringProps = {
+  schema: {
+    type: "object",
+    properties: {
+      version: {
+        type: "number",
+        default: 1
+      },
+      dataSourceInteractive1: {
+        title: "Data Source Interactive 1",
+        type: "string",
+        enum: [],
+        enumNames: []
+      }
+    },
+    // Why this strange chain of dependencies?
+    // Theoretically we could use dataSourceInteractive.type === "array", as react-jsonschema-forms handles that nicely.
+    // But it would make handling of linked interactives difficult. Our hooks / helpers support only top-level
+    // linked interactive properties in the authored state. That way authored state will remain "flat", having just:
+    // .dataSourceInteractive1, ..., .dataSourceInteractive5 properties instead of an array.
```

Comment:
Comment says "5" but we only seem to go up to "3".

Reply:
Thanks, fixed.